server/src/main/java/org/opensearch/extensions/action/ExtensionTransportActionsHandler.java

```diff
@@ -120,19 +120,19 @@ public TransportResponse handleTransportActionRequestFromExtension(TransportActi
                 @Override
                 public void onResponse(ExtensionActionResponse actionResponse) {
                     response.setResponseBytes(actionResponse.getResponseBytes());
-                    inProgressLatch.countDown();
+                    inProgressFuture.complete(actionResponse);
                 }
 
                 @Override
                 public void onFailure(Exception exp) {
                     logger.debug("Transport request failed", exp);
                     byte[] responseBytes = ("Request failed: " + exp.getMessage()).getBytes(StandardCharsets.UTF_8);
                     response.setResponseBytes(responseBytes);
-                    inProgressLatch.countDown();
+                    inProgressFuture.completeExceptionally(exp);
                 }
             }
         );
-        inProgressLatch.await(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
+        inProgressFuture.orTimeout(ExtensionsManager.EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
This one is different than the others; we do `orTimeout()` but we don't do a `get()` on the completion to throw an exception.  Is this intended?